cannon/mipsevm/tests/evm_common_test.go

```diff
@@ -47,10 +47,7 @@ func TestEVM(t *testing.T) {
 				// Short-circuit early for exit_group.bin
 				exitGroup := f.Name() == "exit_group.bin"
 				expectPanic := strings.HasSuffix(f.Name(), "panic.bin")
-
-				evm := testutil.NewMIPSEVM(c.Contracts)
-				evm.SetLocalOracle(oracle)
-				testutil.LogStepFailureAtCleanup(t, evm)
+				validator := testutil.NewEvmValidator(t, c.StateHashFn, c.Contracts, testutil.WithLocalOracle(oracle))
```

Comment:
Cleaned up some of the EVM validation logic so we always do this validation through `testutil` helpers.   Creating this validator object avoids calling `LogStepFailureAtCleanup` at every step, which would be problematic.